Modules/Filtering/ImageFilterBase/include/itkCastImageFilter.hxx

```diff
@@ -163,6 +167,32 @@ CastImageFilter<TInputImage, TOutputImage>::DynamicThreadedGenerateDataDispatche
     inputIt.NextLine();
     outputIt.NextLine();
   }
+#else
+
+  auto inputRange = ImageRegionRange<const TInputImage>(*inputPtr, inputRegionForThread);
+  auto outputRange = ImageRegionRange<TOutputImage>(*outputPtr, outputRegionForThread);
+
+  auto       inputIt = inputRange.begin();
+  auto       outputIt = outputRange.begin();
+  const auto inputEnd = inputRange.end();
+
+
+  OutputPixelType    outputPixel{ *outputIt };
+  const unsigned int componentsPerPixel = NumericTraits<OutputPixelType>::GetLength(outputPixel);
+  while (inputIt != inputEnd)
+  {
+    InputPixelType inputPixel = *inputIt;
```

Comment:
Why not `const InputPixelType & inputPixel`? (The original code also has `const InputPixelType & inputPixel`.)